agent/recognition/binarymatch.py

```
@@ -614,28 +629,38 @@ def _run_preset(self, context: Context,
-            print(f"[RedDotDetector] [preset:{preset_node}] 命中 → {adjusted}")
-            return CustomRecognition.AnalyzeResult(box=adjusted, detail={"preset": preset_node})
-        return None
+            mfaalog.info(f"[RedDotDetector] [preset:{preset_node}] hit -> {adjusted}")
+            return CustomRecognition.AnalyzeResult(
+                box=adjusted, detail={"result": "hit", "preset": preset_node})
+
+        mfaalog.warning(f"[RedDotDetector] miss@preset | {argv.node_name} via {preset_node}")
+        return CustomRecognition.AnalyzeResult(box=None, detail={
+            "result": "miss", "mode": "preset", "preset": preset_node,
+            "roi": [rx, ry, rw, rh],
+            "hint": "阶段原因见预设节点(独立模式)的 detail；失败截图见 debug/RedDotDetector/ 下以本节点名命名的 rdd_* 文件",
```

Ruff (0.15.15)

[warning] String contains ambiguous `；` (FULLWIDTH SEMICOLON). Did you mean `;` (SEMICOLON)?

(RUF001)